pkgs/top-level/all-packages.nix

```diff
@@ -14743,15 +14743,20 @@ with pkgs;
   julia-stable = julia_18;
   julia = julia-stable;
 
-  julia_16-bin = callPackage ../development/compilers/julia/1.6-bin.nix { };
-  julia_18-bin = callPackage ../development/compilers/julia/1.8-bin.nix { };
+  julia_16-bin =  let julia_16-bin' = callPackage ../development/compilers/julia/1.6-bin.nix { };
+                  in callPackage ../development/compilers/julia/with-packages.nix { julia = julia_16-bin'; };
+
+  julia_18-bin = let julia_18-bin' = callPackage ../development/compilers/julia/1.8-bin.nix { };
+                 in callPackage ../development/compilers/julia/with-packages.nix { julia = julia_18-bin'; };
```

Comment:
I did it this way because defining a function which takes a compiler and returns an enhanced compiler avoids having to reorganize all compilers in an elegant way. What's you rationale?

Reply:
Just don't fell like cluttering all-packages.nix with these logic, or  can we extract that into an wrapJulia function>

Reply:
```
  wrapJulia = callPackage ../development/compilers/julia/wrapper.nix { };
  julia_18-bin-unwrapped = callPackage ../development/compilers/julia/1.8-bin.nix { };
  julia_18-bin = wrapJulia julia_18-bin-unwrapped;
```
maybe

Reply:
But you expose to users `wrapJulia` which shouldn't.

Reply:
I think we should expose `wrapJulia`, so that users can use it with custom julia version, or even adapt the wrapper to their needs, just as in `wrapFirefox`

Reply:
OK, `wrapJulia` may have some value, but what about all the `julia...-unwrapped`? It just looks like more clutter to me.

Reply:
passthru the original julia via the wrapper is also fine, I don't have strong opinion on this.